atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/asts/bucketingthings/BucketWriter.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.sweep.asts.bucketingthings;
+
+import com.palantir.atlasdb.sweep.asts.SweepableBucket.TimestampRange;
+import java.util.Optional;
+
+interface BucketWriter {
+    WriteState writeToAllBuckets(
+            long bucketIdentifier,
+            Optional<TimestampRange> oldTimestampRange,
+            TimestampRange newTimestampRange);
+
+    enum WriteState {
```

Comment:
Had a boolean, now an enum to make it clearer. Am happy to just go back to a boolean, or choose a different name